libsyclinterface/CMakeLists.txt

```diff
@@ -44,6 +44,8 @@ option(DPCTL_ENABLE_GLOG
 
 set(DPCTL_LEVEL_ZERO_INCLUDE_DIR "" CACHE STRING "Level zero header path")
 
+set(LIBZE_DEFAULT_LOADER_FILENAME "libze_loader.so.1" CACHE STRING "Default Level Zero Loader filename")
```

Comment:
I'd recommend to guard setting this variable with an `if` and only set it if it isn't already set by the user, like I did [here](https://github.com/sommerlukas/dpctl/commit/c90bc708b7af3b46298de64d9e6d7ac189c4c589). 

This allows users to override the name through CMake options supplied to the build.

Reply:
I considered a similar approach. I think it would be best to add another variable for the user to set instead, because then we can also guard the `readelf` process, since if the user is setting it directly, there's no reason to attempt to inspect for the filename.

Reply:
Added `LIBZE_SET_LOADER_FILENAME` for this purpose